google-ads-examples/src/main/java/com/google/ads/googleads/examples/reporting/ParallelReportDownload.java

```diff
@@ -0,0 +1,253 @@
+      // account.
+      // params.loginCustomerId = Long.parseLong("INSERT_LOGIN_CUSTOMER_ID");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      GoogleAdsClient.Builder builder = GoogleAdsClient.newBuilder().fromPropertiesFile();
+      if (params.loginCustomerId != null) {
+        builder.setLoginCustomerId(params.loginCustomerId);
+      }
+      googleAdsClient = builder.build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new ParallelReportDownload().runExample(googleAdsClient, params.customerIds);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example. This method may block for up to MAX_WAIT_TIME_SECONDS while waiting for
+   * reports to complete.
+   *
+   * @param googleAdsClient the client library instance for API access.
+   * @param customerIds the customer IDs to run against.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, List<Long> customerIds) {
+    // Creates a single client which can be shared by all threads.
+    // gRPC handles multiplexing parallel requests to the underlying API connection.
+    try (GoogleAdsServiceClient serviceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // IMPORTANT: You should avoid hitting the same customer ID in parallel. There are rate limits
+      // at the customer ID level which are much stricter than limits at the developer token level.
+      // Hitting these limits frequently enough will significantly reduce throughput as the client
+      // library will automatically retry with exponential back-off before failing the request.
+      for (String gaqlQuery : GAQL_QUERY_STRINGS) {
+        // Uses a list of futures to make sure that we wait for this report to complete on all
+        // customer IDs before proceeding. The Future data type here is just for demonstration.
+        List<ListenableFuture<ReportSummary>> futures = new ArrayList<>();
+
+        // Calls the API for each customer ID on the current report.
```

Comment:
nit: I think the wording here's a bit confusing. Consider swapping around to something like "Uses the API to retrieve the report for each customer ID".